dts/bindings/usb/st,stm32n6-otghs.yaml

```diff
@@ -0,0 +1,13 @@
+# Copyright (c) 2025, STMicroelectronics
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  STM32N6 OTGHS controller
+
+  In the STM32N6 series, the `pinctrl-0` and `pinctrl-names` properties are not required
+  for the USB OTG HS peripheral configuration. This is because the pin multiplexing
+  for the USB OTG HS peripheral are handled automatically by the hardware.
+
+compatible: "st,stm32n6-otghs"
+
+include: st,stm32-otghs-common.yaml
```

Comment:
@marwaiehm-st @erwango This file is not really necessary. You can just use `compatible = "st,stm32n6-otghs", "st,stm32-otghs";`. In the driver, use first compatible to implement a hook to handle this particular type, what would make your code more maintainable.

Reply:
> This file is not really necessary. 

Are you sure ?
Since this particular IP doesn't make use of pinctrl, it is correct to specify a new compatible that doesn't make `pinctrl` prop required. We can think about making `pinctrl` prop not required in ` "st,stm32-otghs"` compat, but this will reduce the ability to catch wrong configurations and anyway a `"st,stm32n6-otghs"` compat is required to handle IP specifics as you mentioned..

Unless I missed something...

Reply:
> Are you sure ?

Yes, that is how we handle vendor specific hooks in DWC2 driver. There is zero noise in bindings or vendor/type independent code.

Reply:
I'm not sure to understand. You did provide a [`dts/bindings/usb/st,stm32f4-fsotg.yaml`](https://github.com/zephyrproject-rtos/zephyr/blob/main/dts/bindings/usb/st%2Cstm32f4-fsotg.yaml) for `STM32F4 SoC series OTG_FS DWC2 compatible controller` variant.

Reply:
This is to be closer to the scheme the Linux kernel uses for the DWC2 driver.
My point is that something like dts/bindings/usb/st,stm32n6-otghs.yaml without any properties does not make sense, even in the code it looks like a hack

```c
#if !DT_HAS_COMPAT_STATUS_OKAY(st_stm32n6_otghs)
PINCTRL_DT_INST_DEFINE(0);
static const struct pinctrl_dev_config *usb_pcfg =
					PINCTRL_DT_INST_DEV_CONFIG_GET(0);
#endif
```

 I suggest to remove required from the pinctl properties.